utils/dependency_management.py

```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Utility CLI to update dependency version requirements of the `setup.json`."""
+
+import re
+import json
+import copy
+from pathlib import Path
+from collections import OrderedDict
+from pkg_resources import Requirement
+
+import click
+import yaml
+import toml
+
+from validate_consistency import write_setup_json
+
+ROOT = Path(__file__).resolve().parent.parent  # repository root
+
+DEFAULT_EXCLUDE_LIST = ['django', 'circus', 'numpy', 'pymatgen', 'ase', 'monty', 'pyyaml']
+
+SETUPTOOLS_CONDA_MAPPINGS = {
+    'psycopg2-binary': 'psycopg2',
+    'graphviz': 'python-graphviz',
+}
+
+CONDA_IGNORE = [
+    'pyblake2',
+]
+
+
+class DependencySpecificationError(click.ClickException):
+    """Indicates an issue in a dependency specification."""
+
+
+def _load_setup_cfg():
+    """Load the setup configuration from the 'setup.json' file."""
+    try:
+        with open(ROOT / 'setup.json') as setup_json_file:
+            return json.load(setup_json_file)
+    except json.decoder.JSONDecodeError as error:  # pylint: disable=no-member
+        raise DependencySpecificationError("Error while parsing 'setup.json' file: {}".format(error))
+    except FileNotFoundError:
+        raise DependencySpecificationError("The 'setup.json' file is missing!")
+
+
+def _load_environment_yml():
+    """Load the conda environment specification from the 'environment.yml' file."""
+    try:
+        with open(ROOT / 'environment.yml') as file:
+            return yaml.load(file, Loader=yaml.SafeLoader)
+    except yaml.error.YAMLError as error:
+        raise DependencySpecificationError("Error while parsing 'environment.yml':\n{}".format(error))
+    except FileNotFoundError as error:
+        raise DependencySpecificationError(str(error))
+
+
+def _setuptools_to_conda(req):
+    for pattern, replacement in SETUPTOOLS_CONDA_MAPPINGS.items():
+        if re.match(pattern, str(req)):
+            return Requirement.parse(re.sub(pattern, replacement, str(req)))
+    return req  # did not match any of the replacement patterns, just return original
+
+
+@click.group()
+def cli():
+    """Utility to update dependency requirements for `aiida-core`.
+
+    Since `aiida-core` fixes the versions of almost all of its dependencies, once in a while these need to be updated.
```

Comment:
If I understand correctly, what you are referring to here is actually increasing the minimal version requirements of the dependencies.

Might be good to explain when we do this (e.g. before new minor releases / major releases / ...).

To play devil's advocate, one could also argue that such an update actually reduces compatibility of AiiDA with python packages without providing a clear benefit to the user.

Reply:
Same here, all of this does not make any sense anymore, will revise.